bc-backend/src/main/java/org/jboss/da/bc/impl/ProjectHiearchyCreator.java

```diff
@@ -80,16 +82,12 @@ private void iterate(ProjectHiearchy hiearchy) {
         if (!hiearchy.isSelected())
             return; // Not selected, ignoring
 
-        Optional<Set<ProjectHiearchy>> dependencies = hiearchy.getDependencies();
-        if (!dependencies.isPresent())
-            return; // Dependencies were already check with error, ignoring
-
-        if (dependencies.get().isEmpty()) { // Dependencies not yet processed, get and process them
-            GAV gav = hiearchy.getProject().getGav();
-            Optional<GAVToplevelDependencies> deps = getDependencies(gav);
-            hiearchy.setDependencies(toProjectHiearchies(deps));
+        if (hiearchy.getAnalysisStatus().equals(DependencyAnalysisStatus.NOT_ANALYSED)
+                && hiearchy.getDependencies().isEmpty()) {
+            // Dependencies not yet processed, get and process them
+            setDependencies(hiearchy);
         } else { // Dependencies already processed, search next level
-            for (ProjectHiearchy dep : dependencies.get()) {
+            for (ProjectHiearchy dep : hiearchy.getDependencies()) {
```

Comment:
It will work, but still I feel that it would be better to use
```java
}else if (hiearchy.getAnalysisStatus().equals(DependencyAnalysisStatus.ANALYSED)){
```
Or am I just paranoid?